packages/chat-sdk/src/channel/channel.ts

```diff
@@ -983,27 +985,122 @@ export class Channel implements BaseChannel {
     );
   }
 
+  /**
+   * Remove chat cofig changes listener
+   *
+   */
+  private offChatConfigListener(callback?: (item: any) => void, type?: ChatConfigType): void {
+    if (
+      this.chatConfigChangesCallbacks[ChatConfigType.ALL_CHAT_CHANGES].length === 0 &&
+      this.chatConfigChangesCallbacks[ChatConfigType.PIN_MESSAGES].length === 0 &&
+      this.chatConfigListenerUnsubscribe
+    ) {
+      this.chatConfigListenerUnsubscribe();
+      this.chatConfigListenerUnsubscribe = null;
+    }
+
+    if (callback && type) {
+      if (this.chatConfigChangesCallbacks[type].indexOf(callback)) {
+        const newCallbacksList = this.chatConfigChangesCallbacks[type].filter((item) => item !== callback);
+        this.chatConfigChangesCallbacks[type] = newCallbacksList;
+      }
+    }
+  }
```

Comment:
Here we can remove those validations, we only need to filter the array to remove the callback.